internal/gatewayapi/clienttrafficpolicy.go

```diff
@@ -291,6 +291,20 @@ func translateClientTrafficPolicyForListener(policySpec *egv1a1.ClientTrafficPol
 
 		// Translate Proxy Protocol
 		translateListenerProxyProtocol(policySpec.EnableProxyProtocol, httpIR)
+		// enable http3 if set and TLS is enabled
+		if httpIR.TLS != nil && policySpec.HTTP3 != nil {
+			httpIR.HTTP3 = &ir.HTTP3Settings{}
+			var proxyListenerIR *ir.ProxyListener
+			for _, proxyListener := range infraIR[irKey].Proxy.Listeners {
```

Comment:
nit: can we reuse `irListenerName` and also check if `irListenerName.Proxy != nil`